packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -211,8 +211,14 @@ class CurlyComponentManager {
     component.trigger('didInitAttrs', { attrs });
     component.trigger('didReceiveAttrs', { newAttrs: attrs });
 
-    if (environment.isInteractive) {
-      component.trigger('willRender');
+    // We usually do this in the `didCreateElement`, but that hook doesn't fire for tagless components
```

Comment:
I'd like to move this logic to a central location (to avoid duplicating the exact order and conceptual logic both here and in `didCreateElement`).  In a similar situation, I added logic to the `ComponentStateBucket` object but it could likely live in a few different places based on your preferences.